registry/repository.go

```diff
@@ -55,7 +56,7 @@ func (c *RepositoryClient) ListSignatureManifests(ctx context.Context, manifestD
 	// TODO(shizhMSFT): filter artifact type at the server side
 	if err := c.Repository.Referrers(ctx, ocispec.Descriptor{
 		Digest: manifestDigest,
-	}, func(referrers []artifactspec.Descriptor) error {
+	}, ArtifactTypeNotation, func(referrers []artifactspec.Descriptor) error {
```

Comment:
Just a question. Since we have already provided `ArtifactTypeNotation` to the refer api. do we still need to manually filter it again in line 61?

Reply:
https://github.com/oras-project/oras-go/blob/97a9c43c52f9d89ecf5475bc59bd1f96c8cc61f6/registry/remote/repository.go#L455-L481
I think api will filter it if we pass the artifact type.

Reply:
Good call, it can be removed safely now.

Reply:
I think client side filtering should still be retained as `artifactType` filtering is an optional server feature as per ORAS [referrers spec](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md#filtering-results). I'm not sure though that the response contains any indication from the server that `artifactType` filtering was not applied.

@shizhMSFT any feedback?

Reply:
@gokarnm The client-side filtering is ensured by the `oras-go` SDK.